core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -205,6 +207,15 @@ void setSpecId(int specId) {
     this.partitionSpecId = specId;
   }
 
+  @Override
+  public Long sequenceNumber() {
+    return sequenceNumber;
+  }
+
+  void setSequenceNumber(Long seq) {
```

Comment:
Sequence number is tracked by the manifest entry, not the content file. And I don't think it is a good idea to make this mutable. What is the rationale for adding it here (not just a ref to another issue)?

Reply:
I was following the specId change mentioned in https://github.com/apache/iceberg/issues/4749.

Reply:
The spec ID isn't available in manifest entry, which is why we thought it was reasonable to add it. Since this is available in the manifest entry, I would prefer to use it from there. That is, unless this is blocking something and we _need_ to add it for some reason.

Reply:
This is motivated when writing https://github.com/apache/iceberg/pull/4748 which handles sequence numbers. And I think we should need this when cleaning the delete files in the future.  Right now, we don't clean the delete files after bin-packing.